From: Yifeng <wuyifeng@nd.com.cn>
Date: Tue, 2 Aug 2022 07:40:00 +0000
Subject: Change default user agent

Fix version: 1.2.0
Add `XiaoMi/MiuiBrowser/16.2.16 swan-mibrowser` to the android user agent.
Hide the `Desktop site` check box when the page is `chrome://` or `chrome-native://`.

---
 chrome/android/java/src/org/chromium/chrome/browser/app/appmenu/AppMenuPropertiesDelegateImpl.java | 2 +-
 content/common/user_agent.cc | 2 +-
 content/public/common/user_agent.h | 2 +-
 3 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/chrome/android/java/src/org/chromium/chrome/browser/app/appmenu/AppMenuPropertiesDelegateImpl.java b/chrome/android/java/src/org/chromium/chrome/browser/app/appmenu/AppMenuPropertiesDelegateImpl.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/app/appmenu/AppMenuPropertiesDelegateImpl.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/app/appmenu/AppMenuPropertiesDelegateImpl.java
@@ -1188,7 +1188,7 @@ public class AppMenuPropertiesDelegateImpl implements AppMenuPropertiesDelegate

         // Hide request desktop site on all chrome:// pages except for the NTP.
         boolean itemVisible = currentTab != null && canShowRequestDesktopSite
-                && (!isChromeScheme || currentTab.isNativePage())
+                && !isChromeScheme
                 && !shouldShowReaderModePrefs(currentTab) && currentTab.getWebContents() != null;
         requestMenuRow.setVisible(itemVisible);
         if (!itemVisible) return;
diff --git a/content/common/user_agent.cc b/content/common/user_agent.cc
--- a/content/common/user_agent.cc
+++ b/content/common/user_agent.cc
@@ -336,7 +336,7 @@ std::string BuildUserAgentFromOSAndProduct(const std::string& os_info,
   std::string user_agent;
   base::StringAppendF(&user_agent,
                       "Mozilla/5.0 (%s) AppleWebKit/537.36 (KHTML, like Gecko) "
-                      "%s Safari/537.36",
+                      "%s Safari/537.36 Gecko/20100101 Firefox/103.0",
                       os_info.c_str(), product.c_str());
   return user_agent;
 }
diff --git a/content/public/common/user_agent.h b/content/public/common/user_agent.h
--- a/content/public/common/user_agent.h
+++ b/content/public/common/user_agent.h
@@ -21,7 +21,7 @@ const char kDesktop[] =
 const char kAndroid[] =
     "Mozilla/5.0 (%s) "
     "AppleWebKit/537.36 (KHTML, like Gecko) Chrome/%s.0.0.0 %s"
-    "Safari/537.36";
+    "Safari/537.36 Gecko/20100101 Firefox/103.0";
 const char kUnifiedPlatformAndroid[] = "Linux; Android 10; K";
 const char kUnifiedPlatformCrOS[] = "X11; CrOS x86_64 14541.0.0";
 const char kUnifiedPlatformLinux[] = "X11; Linux x86_64";
--